protocol/directory_test.go

```diff
@@ -138,3 +139,231 @@ func TestRegisterExistedUserWithoutTB(t *testing.T) {
 		t.Fatal("Expect returned TB is nil")
 	}
 }
+
+func TestKeyLookupWithTB(t *testing.T) {
+	d := newDirectory(t, true)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence and the TB of looking up user
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if res.Type != KeyLookupType {
+		t.Fatal("Expect response type", KeyLookupType, "got", res.Type)
+	}
+	if res.Error != Success {
+		t.Fatal("Expect no error", "got", res.Error)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB == nil || !bytes.Equal(res.TB.Value, []byte("key")) {
+		t.Fatal("Expect the TB of looking up user")
+	}
+
+	d.Update()
+	// lookup in epoch after registering epoch
+	// expect a proof of inclusion
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if ap := res.AP; ap == nil || !bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of inclusion")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupNameNotFoundWithTB(t *testing.T) {
+	d := newDirectory(t, true)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "bob"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in epoch after registering epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "bob"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupWithoutTB(t *testing.T) {
```

Comment:
How is this test different from `TestKeyLookupWithTB`? Also, the lookup in same epoch in that test above would also return an `ErrorNameNotFound`, right?